odoo/addons/base/models/res_lang.py

```diff
@@ -209,77 +240,64 @@ def install_lang(self):
                 partner.write({'lang': lang_code})
         return True
 
-    @tools.ormcache('code')
-    def _lang_get_id(self, code):
-        return self.with_context(active_test=True).search([('code', '=', code)]).id
-
-    @tools.ormcache('code')
-    def _lang_get_direction(self, code):
-        return self.with_context(active_test=True).search([('code', '=', code)]).direction
-
-    @tools.ormcache('url_code')
-    def _lang_get_code(self, url_code):
-        return self.with_context(active_test=True).search([('url_code', '=', url_code)]).code or url_code
+    # ------------------------------------------------------------
+    # cached methods for **active** languages
+    # ------------------------------------------------------------
+    @property
+    def CACHED_FIELDS(self) -> list[str]:
+        """ Return the fields to cache for the active languages
+        Please make sure all these fields don't depend on other models or context and are not translated
+        """
+        return ['id', 'name', 'code', 'iso_code', 'url_code', 'active', 'direction', 'date_format', 'time_format',
+                'week_start', 'grouping', 'decimal_point', 'thousands_sep', 'flag_image_url']
+
+    def _get_data(self, **kwargs: Any) -> LangData:
+        """ Get the language data for the **first** given field value in kwargs
+        For example, get_data(code='en_US') will return the LangData
+        for the res.lang record whose 'code' field value is 'en_US'
+
+        :param dict kwargs: {field_name: field_value, ...}
+                field_name is the first key in kwargs and can be any self.CACHED_FIELDS
+                Try to reuse the already used 'field_name's: 'id', 'code', 'url_code'
+        :return: Valid LangData if field_name, field value pair is for an **active** language
+                Otherwise, Dummy LangData which will return `False` for any field
+        :rtype: LangData
+        :raise: KeyError if field_name is not in self.CACHED_FIELDS
+        """
+        field_name, field_value = next(iter(kwargs.items()))
+        return self._get_active_by(field_name)[field_value]
 
-    def _lang_get(self, code):
+    def _lang_get(self, code: str):
         """ Return the language using this code if it is active """
-        return self.browse(self._lang_get_id(code))
+        return self.browse(self._get_data(code=code).id)
 
-    def _get_active(self, code):
-        """ Return the given language code if active, else return ``"en_US"``.
+    def _get_valid(self, code: str) -> str:
```

Comment:
Since we are moving routing methods from base to routing, shouldn't we move `_get_valid` in the only module using it (l10n_gcc_invoice)?